src/components/Experiments/ExperimentEditModal.js

```diff
@@ -0,0 +1,195 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  Button,
+  Modal,
+  FormGroup,
+  FormControl,
+  Glyphicon,
+  Row,
+  Col,
+} from 'react-bootstrap';
+
+import {
+  showEditModal,
+  create as experimentCreate,
+  update as experimentUpdate,
+  uploadFilePairs,
+  UPLOAD_RES_SUCCESS,
+  UPLOAD_RES_FAILURE,
+} from '../../state/experiments';
+
+import './ExperimentEditModal.less';
+
+class ExperimentEditModal extends Component {
+  constructor(props) {
+    super(props);
+
+    const exp = this.props.editModalExperiment;
+    if (exp !== undefined) {
+      this.state = { nameVal: exp.name, descriptionVal: exp.description };
+    } else {
+      this.state = {
+        nameVal: '',
+        descriptionVal: '',
+      };
+    }
+
+    this.handleClose = this.handleClose.bind(this);
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleUpload = this.handleUpload.bind(this);
+  }
+
+  handleClose() {
+    this.props.showEditModal(false);
+  }
+
+  handleNameChange(e) {
+    this.setState({ nameVal: e.target.value });
+  }
+
+  handleDescriptionChange(e) {
+    this.setState({ descriptionVal: e.target.value });
+  }
+
+  handleSave() {
+    const create = this.props.editModalExperiment === undefined;
+
+    if (create) {
+      this.props.experimentCreate(
+        this.state.nameVal,
+        this.state.descriptionVal
+      );
+    } else {
```

Comment:
please follow `early return` rule as all other js code.

Reply:
Changed in https://github.com/src-d/code-annotation/pull/195/commits/0fb33e0f998609b4e0e9304e7576194fb695b0ad

Can you please point me to a guideline (like the airbnb one we use in eslint.rb) where this rule is explained?

Reply:
the rule usually you look how the current code is written and do the same.

there is a rule about no-else return in airbnb:
https://github.com/airbnb/javascript#blocks--no-else-return

no **always** early return because sometimes it makes sense to not return early (though it was there before)

you can read more why people do that by simple googling `early return`. It's very popular pattern in javascript. Java style of `else` everywhere is much less common.

Reply:
Ok, thanks for clarifying. I'm aware of the rules you mention, but I just didn't see them applying here and wanted to check if you were talking about something else. From my point of view the code was:
```
if (condition)
    somehting()
else
    other()
end

// Posibly some more code
```

Now I understand your point of view, as there was no more code after the else.